### ISSUE TITLE
Is it possible to get the phone screen that I have with me  to a remote virtual machine that is on a different network? 

### ISSUE THREAD

**Issue by vladpikaciu** [Genymobile/scrcpy#1776]
So I want to stream the screen of my phone to a Windows VM from Azure that is not on or near my network. Is there a way I could do this? I am still a beginner so I don't know how this could be done..

**Comment by janosvitok** [comment]
[I don't know how you connect to Azure VM.]

# Approach 1

https://github.com/Genymobile/scrcpy#ssh-tunnel

You'll need a computer in a local network. Connect your phone to that computer using USB, create [reverse SSH tunnel](https://www.google.com/search?q=reverse+ssh+tunnel) to your Azure VM, and then forward adb connections from Azure to adb running on your local computer as described in the above paragraph.

You may use other VPN if ssh is not available.

If that works, you may try connecting the phone using WIFI if you want to get rid of the USB cable.

# Approach 2

You can run scrcpy on the local computer and use remote desktop/team viewer/skype screensharing to connect from azure VM to your local computer.

# Approach 3

Run scrcpy locally, stream local window using VLC.

**Comment by igolf-plat** [comment]
What if both computers are behind nat device ? Could it be done?

**Comment by rom1v** [comment]
> What if both computers are behind nat device ? Could it be done?

This could be done only if you both can reach a third computer.

**Comment by igolf-plat** [comment]
> > What if both computers are behind nat device ? Could it be done?
> 
> This could be done only if you both can reach a third computer.

Yes, I can ssh into android device through a third computer by forwarding ssh port. But when I forward 5555 port , adb connect fails. I dont know why.
